backend/inspirehep/matcher/config.py

```diff
@@ -334,3 +334,232 @@
+            ]
+        }
+    ],
+    "index": "records-hep",
+}
+
+EXPERIMENT_REFERENCE_IN_AUTHOR_RECORD_CONFIG = {
+    "algorithm": [
+        {
+            "queries": [
+                {
+                    "path": "$ref",
+                    "search_path": "project_membership.record.$ref",
+                    "type": "exact",
+                },
+            ]
+        }
+    ],
+    "index": "records-authors",
+}
+
+EXPERIMENT_REFERENCE_IN_JOBS_RECORD_CONFIG = {
+    "algorithm": [
+        {
+            "queries": [
+                {
+                    "path": "$ref",
+                    "search_path": "accelerator_experiments.record.$ref",
+                    "type": "exact",
+                },
+            ]
+        }
+    ],
+    "index": "records-jobs",
+}
+
+CONFERENCE_REFERENCE_IN_LIT_RECORD_CONFIG = {
+    "algorithm": [
+        {
+            "queries": [
+                {
+                    "paths": ["$ref"],
+                    "search_paths": ["publication_info.conference_record.$ref"],
+                    "type": "nested",
+                },
+            ]
+        },
+    ],
+    "index": "records-hep",
+}
+
+JOURNAL_REFERENCE_IN_LIT_RECORD_CONFIG = {
+    "algorithm": [
+        {
+            "queries": [
+                {
+                    "paths": ["$ref"],
+                    "search_paths": ["publication_info.journal_record.$ref"],
+                    "type": "nested",
+                },
+            ]
+        },
+    ],
+    "index": "records-hep",
+}
+
+
+INSTITUTION_REFERENCE_IN_LIT_RECORD_AUTHORS_CONFIG = {
+    "algorithm": [
+        {
+            "queries": [
+                {
+                    "paths": ["$ref"],
+                    "search_paths": ["authors.affiliations.record.$ref"],
+                    "type": "nested",
+                },
+            ]
+        },
+    ],
+    "index": "records-hep",
+}
+
+
+INSTITUTION_REFERENCE_IN_LIT_RECORD_THESIS_INFO_CONFIG = {
+    "algorithm": [
+        {
+            "queries": [
+                {
+                    "path": "$ref",
+                    "search_path": "thesis_info.institutions.record.$ref",
+                    "type": "exact",
+                },
+            ]
+        },
+    ],
+    "index": "records-hep",
+}
+
+INSTITUTION_REFERENCE_IN_AUTHOR_RECORD_CONFIG = {
+    "algorithm": [
+        {
+            "queries": [
+                {
+                    "path": "$ref",
+                    "search_path": "positions.record.$ref",
+                    "type": "exact",
+                },
+            ]
+        }
+    ],
+    "index": "records-authors",
+}
+
+INSTITUTION_REFERENCE_IN_JOB_RECORD_CONFIG = {
+    "algorithm": [
+        {
+            "queries": [
+                {
+                    "path": "$ref",
+                    "search_path": "institutions.record.$ref",
+                    "type": "exact",
+                },
+            ]
+        }
+    ],
+    "index": "records-jobs",
+}
```

Comment:
this seems like a terrible idea to me, yet another place to update when the schema changes. Maybe we could move this information to the schemas somehow and generate these searches based on this info?